### ISSUE TITLE
Breite-Light.ufo refers to “component_1” without defining it

### ISSUE THREAD

**Issue by brawer** [metapolator/Breite#1]
In Breite-Light, glyph uni090E [refers](https://github.com/metapolator/Breite/blob/master/Breite%20Light/Light.ufo/glyphs/uni090E_.glif#L7) to “component_1” which is not defined by the font. Because of this dangling reference, [fontmake](https://github.com/googlei18n/fontmake) cannot build the font.

**Comment by graphicore** [comment]
This reminds me of a finding I made recently: https://github.com/Tarobish/Katibeh/issues/94#issuecomment-181968957

> I was looking in the ufos exported by Glyphs.
> Glyphs doesn't translate all of the component names correctly into production names when exporting the ufo: 
> …

**Comment by graphicore** [comment]
> This reminds me …

But this is different, there's no other glyph that could be `component_1` with a mangled name.

@brawer I am wondering, while the UFO docs say nothing about how to handle missing components, the pen protocol is explicit in skipping them, see: https://github.com/behdad/fonttools/blob/master/Lib/fontTools/pens/basePen.py#L197 If you would use a normal segment pen in fontmake, it should just drop the missing component without further notice. This is may be the reason why other ufo-consumers did not complain earlier about such files. But, I'm not aware of each component handling case in ufoLib. Is the source of your error in ufoLib? I guess the source of the UFO here is Glyphs, so someone should have noticed this before—well, or not.

**Comment by brawer** [comment]
Just double-checking before nagging the Glyphs folks: can you confirm that [Breite-Light.ufo](https://github.com/metapolator/Breite/tree/master/Breite%20Light/Light.ufo) was indeed written by Glyphs?

**Comment by graphicore** [comment]
Good call, and I don't know in fact :-) @nicolasfpauly how did you make these ufos?

Seems, like I was wrong BTW:
https://github.com/metapolator/Breite/blob/master/Breite%20Light/Light.ufo/metainfo.plist#L6

```
<key>creator</key>
<string>org.robofab.ufoLib</string>
```

While a recent ufo generated by glyphs:
https://github.com/Tarobish/Mirza/blob/master/Generated/Arabic/Mirza-Regular.ufo/metainfo.plist#L6

```
    <key>creator</key>
    <string>com.schriftgestaltung.GlyphsUFOExport</string>
```
